azerothcore/azerothcore-wotlk
# fix(Core): Missing updates for creatures; disable Travis modules check

##### CHANGES PROPOSED:
This PR fixes a problem introduced with PR #2304. Sadly I had a logic error in there: Cells visited only for large creatures were also marked as visited for all creatures, which is wrong because it prevents updates for those creatures. With this PR cells are marked as visited for large and normal creatures separately which ensures that each creature is updated.

Sadly I did not see this error in the first place as it only shows while playing with multiple players.

This PR will also temporarily disable the Travis modules check in order to allow correct Travis runs once more. This will be reverted after PR #2454 has been merged and the modules have been fixed.

##### ISSUES ADDRESSED:
- Closes #2390
- Closes #2399

##### TESTS PERFORMED:
- Tested build on Ubuntu 16.04 x64 / cmake 3.15.5 / clang 7.1.0 / MySQL 5.7
- Tested successfully in-game

##### HOW TO TEST THE CHANGES:
Without this PR:
- Create 2 hunters on separate accounts, put their pets in "defensive" mode
- `.go 7343.97 -896.434 915.94 571`
- Attack one of the rhinos there with the pet of the first hunter
- Attack the same rhino with the pet of the second hunter
- Run away with the second hunter, constantly attacking the rhino with the pet, so it keeps following
- One of the pets will stop getting updates, which means it "hangs" or "lags", unresponsive to the player's commands

With this PR the pets should just behave normally.

##### KNOWN ISSUES AND TODO LIST:
none

##### Target branch(es):
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
